frostsnap_core/src/coordinator.rs

```diff
@@ -351,18 +350,25 @@ impl FrostCoordinator {
                     let signatures = sessions
                         .iter()
                         .map(|session_progress| {
-                            let xonly_frost_key =
-                                session_progress.key.into_frost_key().into_xonly();
+                            let xonly_frost_key = session_progress.tweaked_frost_key();
 
-                            frost.combine_signature_shares(
+                            let sig = frost.combine_signature_shares(
                                 &xonly_frost_key,
                                 &session_progress.sign_session,
                                 session_progress
                                     .signature_shares
                                     .iter()
                                     .map(|(_, &share)| share)
                                     .collect(),
-                            )
+                            );
+
+                            assert!(session_progress.sign_item.verify(
+                                &Schnorr::verify_only(),
+                                session_progress.key.into_frost_key().public_key(),
```

Comment:
There's something odd here. Why do we call `session_progress.sign_item.verify` and then pass in another `session_progress` thing into the call. It looks like `.verify` should just be on `session_progress`.